addOns/automation/src/main/java/org/zaproxy/addon/automation/DefaultPropertyFilter.java

```diff
@@ -35,11 +34,8 @@ public void serializeAsField(
             Object pojo, JsonGenerator jgen, SerializerProvider provider, PropertyWriter writer)
             throws Exception {
         if (include(writer)) {
-            Object value = writer.getMember().getValue(pojo);
-            if (value == null
-                    || (value instanceof Collection && ((Collection<?>) value).isEmpty())
```

Comment:
We wouldn't need these checks if defaults are set correctly for all jobs. Also, in cases where the default is a non-null value or a non-empty list, this would prevent the user from overriding such fields.